### PR TITLE
Fixes consistency issue

### PR DESCRIPTION
**Pull request by naved001**
* This handles a case where we can get registration failed error due to
permission issues during provision. Allows to roll-back the database to the
correct state.
* may not work if the roll back process itself crashses; @chemistry-sourabh has
a more, hopefully, robust PR that promises to take care of that.

### CODE REVIEW

File: ims/einstein/operations.py [hunk 1]
```diff
@@ -216,6 +214,8 @@ def shutdown(self):
     @log
     def provision(self, node_name, img_name, network, nic):
         try:
+            mac_addr = ("01-" + self.hil.get_node_mac_addr(node_name).
+                        replace(":", "-"))
```
Comment (chemistry-sourabh) [inline ims/einstein/operations.py]: Lol remove extra set of parathesis

Comment (naved001) [inline ims/einstein/operations.py]: I added that to split it for pep8. Otherwise I don't know how to split that line. want me to use `\`?

Comment (chemistry-sourabh) [inline ims/einstein/operations.py]: Better

Comment (naved001) [inline ims/einstein/operations.py]: done

**Comment by naved001** [comment]
@ravisantoshgudimetla There's no test case to be updated. Though I can write one, but I am not sure how I would do that because what I am fixing here only happens when BMI isn't run as root (which leads to IOError or OSError which this handles). 

If you have ideas to write something to test that scenario, then I'll do that otherwise I think it's good.

**Comment by ravisantoshgudimetla** [comment]
Usually, you have to fake it for unit test. But we don't have one yet, easiest would be to update the existing deployment tests that @chemistry-sourabh  wrote some time back, if possible. I just looked at the code, it seems we don't have -ve scenario's being tested. This could be an addition. If not, leave it as it with a TODO, saying that this has to be tested.

**Comment by naved001** [comment]
Given how sporadic my contributions have been, I have decided to do the latter.

**Comment by chemistry-sourabh** [comment]
Better do the later as all the negative test cases, can be written at once

**Comment by naved001** [comment]
@ravisantoshgudimetla made the issue. 

We can just use the squash and merge button :)

**Comment by chemistry-sourabh** [comment]
@naved001 you can do the honors

**Pull request merged**